core/src/main/java/org/primefaces/extensions/component/keynote/KeynoteRenderer.java

```diff
@@ -0,0 +1,296 @@
+
+import javax.faces.FacesException;
+import javax.faces.application.Resource;
+import javax.faces.component.UIComponent;
+import javax.faces.context.ExternalContext;
+import javax.faces.context.FacesContext;
+import javax.faces.context.ResponseWriter;
+
+import org.primefaces.extensions.model.keynote.KeynoteItem;
+import org.primefaces.extensions.util.Attrs;
+import org.primefaces.renderkit.CoreRenderer;
+import org.primefaces.util.LangUtils;
+import org.primefaces.util.WidgetBuilder;
+
+public class KeynoteRenderer extends CoreRenderer {
+
+    public static final String CONTAINER_CLASS = "ui-keynote reveal";
+    public static final String SLIDES_CLASS = "slides";
+    public static final String ITEM_CLASS = "ui-keynote-item";
+    public static final String SPEAKER_NOTE_CLASS = "notes";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void decode(final FacesContext context, final UIComponent component) {
+        decodeBehaviors(context, component);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void encodeBegin(FacesContext context, UIComponent component) throws IOException {
+        final Keynote keynote = (Keynote) component;
+
+        // encodeCSS(context, "primefaces-extensions", "keynote/theme/" + keynote.getTheme() + ".css");
+        // encodeCSS(context, "primefaces", "primeicons/primeicons.css");
+        if (keynote.getTheme() != null && !"none".equals(keynote.getTheme())) {
+            encodeCSS(context, keynote.getLibrary(), keynote.getTheme());
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void encodeEnd(final FacesContext context, final UIComponent component) throws IOException {
+        final Keynote keynote = (Keynote) component;
+
+        encodeMarkup(context, keynote);
+        encodeScript(context, keynote);
+    }
+
+    /**
+     * Create the HTML markup for the DOM.
+     */
+    private void encodeMarkup(final FacesContext context, final Keynote keynote) throws IOException {
+        final ResponseWriter writer = context.getResponseWriter();
+        final String clientId = keynote.getClientId(context);
+        final String styleClass = getStyleClassBuilder(context)
+                    .add(CONTAINER_CLASS)
+                    .add(keynote.getStyleClass())
+                    .build();
+
+        writer.startElement("div", keynote);
+        writer.writeAttribute("id", clientId, null);
+        writer.writeAttribute(Attrs.CLASS, styleClass, "styleClass");
+        if (keynote.getStyle() != null) {
+            writer.writeAttribute(Attrs.STYLE, keynote.getStyle(), Attrs.STYLE);
+        }
+
+        writer.startElement("div", null);
+        writer.writeAttribute(Attrs.CLASS, SLIDES_CLASS, "class");
+
+        if (keynote.getVar() != null) {
+            // dynamic items
+            final Object value = keynote.getValue();
+            if (value != null) {
+                if (!(value instanceof Collection<?>)) {
```

Comment:
This can simply be
```suggestion
                if (!(value instanceof Collection)) {
```
right?